crates/datastore/src/locking_tx_datastore/mut_tx.rs

```diff
@@ -1947,6 +1952,11 @@ impl MutTxId {
     /// The lock on the committed state is converted into a read lock,
     /// and returned as a new read-only transaction.
     ///
+    /// IMPORTANT: This method updates the in-memory state of the database but does not make it durable.
+    /// That is, the tx will not be persisted to the commitlog.
+    /// Hence you should be careful when calling this method directly.
+    /// In most cases you'll want to use `RelationalDB::commit_tx_downgrade` which makes the tx durable.
+    ///
```

Comment:
Can't mention exact line as its not in diff but just below can you make `pub fn commit_downgrade`  to be `pub(super)` (like ` fn commit`)? That would have prevented this bug.